### ISSUE TITLE
Collaborate with me?

### ISSUE THREAD

**Issue by billiegoose** [pladaria/create-node-module#1]
I was looking on npm to see if the name "create-node-module" was available for my [boilerplate script 'cnm'](https://github.com/wmhilton/cnm/blob/master/cli.sh) (which stands for Create Node Module), when I found yours already has that name and does almost the same things! Mine is written in shell script too, and we both automate creating the repo on Github, filling out package.json and README.md, adding ava for tests and Travis for CI. The main difference is yours uses a config file whereas mine reads from .git/config and .npmrc. Adding a config file was on my TODO list.

I'm about to sit down and write a new version of my boilerplate generator in JavaScript so it's more portable, and wondered if that was something you're interested in too? I could add in your features (or we could make them configurable) and then since you've got the awesome name on npm, you could publish it as version 2.0. And then we can try to make it as popular as `create-react-app`!

**Comment by pladaria** [comment]
Hi, 
Sounds cool, I also wanted to port it to JS and add new features. 
Send me a PR once you have the basic structure so we can take that as starting point and discuss the organization and features.

**Comment by billiegoose** [comment]
Just want to followup to say this will happen! I just get busy. I did start on a basic structure a while ago, but only got as far as a command line program that prints a help and validates options. But since that's all I've got, I might as well show you a little instead of nothing.

https://github.com/pladaria/create-node-module/compare/master...wmhilton:js

It's enough of a start that there's some things we can disagree about, like what command line parsers, or async/await, etc. You might really like it or you might be completely repulsed and say, "You don't use semicolons? That's it, we can't be friends anymore". ¯\\\_(ツ)\_/¯

**Comment by pladaria** [comment]
Hi,

Looks good! I'm also very busy so I wish I had time to properly review the
code, maybe next week (I have some free days).

I like async/await... but I also like 4 spaces and semicolons xD

To not start a dumb fight we can use https://github.com/prettier/prettier
with default settings to format the code. What do you think?

Best regards!

2017-04-09 6:32 GMT+02:00 William Hilton <notifications@github.com>:

**Comment by billiegoose** [comment]
... FOUR spaces? 🤣 Yeah, let's use `prettier`. I use `standard` myself, but am unashamedly pragmatist.

Glad you like async/await! The more I use it the more I miss it when I have to go without it.

**Comment by billiegoose** [comment]
Come to think of it, I'll run the code through prettier before you read it. :)

**Comment by billiegoose** [comment]
I don't know WHAT I was smoking when I wrote that first attempt at code. Actually I do know, I was trying to design from the top down, which never works. The new code is MUCH more concise, and actually does something: generate a .gitignore from the Github templates!

![image](https://cloud.githubusercontent.com/assets/587740/24840987/468f8dae-1d47-11e7-97de-adacedfd4d91.png)

So what I've got now is basically organized like this:
- `src`
  - `bin`
    - `cli.js` - putting the CLI bits here allows /src/index.js to just expose the API for other modules to use
  - `index.js` - executes each feature one after the other
  - `gitignore.js` - the gitignore feature
  - `...` - travis feature, github feature, etc each "feature" gets a file for easy testing
- `dist` - same exact structure as /src but transpiled
